### PR TITLE
Fix layout insertion cwf

### PR DESCRIPTION
**Pull request by wxwxwxwx9**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**
• [x] Bug fix

**What is the rationale for this request?**
As I was working on issue #824, I discovered a bug where the current working file (cwf) is not set for layout insertions. This PR fixes the bug. 

**What changes did you make? (Give an overview)**
Wrapped the return value of the layout insertion functions with a `div` containing the correct `cwf`. 

**Testing instructions:**
`npm run test`

**Proposed commit message: (wrap lines at 72 characters)**
Fix layout insertion cwf

### CODE REVIEW

File: packages/cli/test/functional/test_site/expected/testThumbnails.html [hunk 1]
```diff
@@ -46,8 +46,8 @@
           <thumbnail src="http://placehold.it/100x300" circle></thumbnail>
         </p>
         <p><strong>An actual image</strong>
-          <thumbnail circle size="300" src="/test_site/images/deer.jpg"></thumbnail>
-          <pic src="/test_site/images/deer.jpg" height="300"></pic>
+          <thumbnail circle size="300" src="/test_site/_markbind/layouts/default/images/deer.jpg"></thumbnail>
+          <pic src="/test_site/_markbind/layouts/default/images/deer.jpg" height="300"></pic>
```
Comment (wxwxwxwx9) [inline packages/cli/test/functional/test_site/expected/testThumbnails.html:50]: Not sure why the test updates are broken!

File: packages/core/src/Page/index.js [hunk 2]
```diff
@@ -586,7 +586,7 @@ class Page {
 
     const renderedFooter = this.pageConfig.variableProcessor.renderSiteVariables(this.pageConfig.sourcePath,
                                                                                  footerContent, pageSources);
-    return `${pageData}\n${renderedFooter}`;
+    return `<div data-included-from="${this.footer}">${pageData}</div>\n${renderedFooter}`;
```
Comment (ang-zeyu) [inline packages/core/src/Page/index.js:589]: 👀

Comment (ang-zeyu) [reply packages/core/src/Page/index.js:589]: let's add a test or two too: any relative link in one of the test_site layouts

Comment (wxwxwxwx9) [reply packages/core/src/Page/index.js:589]: Ah, this was very careless of me. Thanks for spotting this!

File: packages/core/src/Page/index.js [hunk 3]
```diff
@@ -567,7 +567,7 @@ class Page {
 
     const renderedHeader = this.pageConfig.variableProcessor.renderSiteVariables(this.pageConfig.sourcePath,
                                                                                  headerContent, pageSources);
-    return `${renderedHeader}\n${pageData}`;
```
Comment (ang-zeyu) [inline packages/core/src/Page/index.js:570]: the newlines are important to ensure the different segments get their markdown rendered correctly

**Review by ang-zeyu — COMMENTED** [review]
thanks! let's add tests too (something like relative links that relies on cwf would do)

File: packages/core/src/Page/index.js (hunk 3 above)
Comment (wxwxwxwx9) [reply packages/core/src/Page/index.js:570]: Ah, I see! Have added the newline back to the code.

**Comment by wxwxwxwx9** [comment]
Hi Zeyu, I have added a test for the bug fix. Let me know if it's okay and whether there are other things that I should add! Thanks so much :-)

File: packages/cli/test/functional/test_site/testRelativeLink.md [hunk 4]
```diff
@@ -0,0 +1 @@
+**Relative Link Test** This is a relative Intra-Site link (see [Tags](testTags.md))
```
Comment (ang-zeyu) [inline packages/cli/test/functional/test_site/testRelativeLink.md:1]: let's move this to one of the layout files instead, which is the use case you fixed here

Comment (wxwxwxwx9) [reply packages/cli/test/functional/test_site/testRelativeLink.md]: Updated the test! But I'm not sure if it is the most appropriate. Let me know if I should make more changes!

File: packages/cli/test/functional/test_site/_markbind/layouts/default/head.md [hunk 5]
```diff
@@ -2,3 +2,5 @@
 <meta name="default-head-top">
 </head-top>
 <meta name="default-head-bottom">
+
+**Relative Link Test** This is a relative Intra-Site link (see [Tags](../../../testTags.html))
```
Comment (ang-zeyu) [inline packages/cli/test/functional/test_site/_markbind/layouts/default/head.md:6]: you want `header.md`, the similar sounding `head.md` is for inserting things into the `<head>` tag =P

**Review by ang-zeyu — APPROVED** [review]
_Lgtm_

**Pull request merged**